### DISCUSSION TITLE
Does anybody know the current state of MTLMultisampleDepthResolveFilter on metal?

### DISCUSSION THREAD

**Discussion by FilippoLeon** [KhronosGroup/MoltenVK#1274] [Q&A]
As far as I understand, MoltenVK seems touse "macOS_GPUFamily2_v1" to enable the availability to eMin and eMax.

I cannot find enough documentation on Metal to verify that this is in fact the case. All I found is whether "depth resolve" is supported at all. There also seem no way to query support for those features directly from metal.

I still have to test it, but it seems to me that on some GPUs, eMin and eMax are either not available, or not working, even tough they belong to GPU family 2 (I use a mac book pro 2016, which is barely family 2, but also an iMac 2020, which is definitely well within family 2).

On a related note: does anybody know of something better than this (and the associated pdf):
https://developer.apple.com/documentation/metal/mtlfeatureset

to figure out various capabilities of metal.

**Comment by cdavis5e** [comment]
> I still have to test it, but it seems to me that on some GPUs, eMin and eMax are either not available, or not working, even tough they belong to GPU family 2 (I use a mac book pro 2016, which is barely family 2, but also an iMac 2020, which is definitely well within family 2).

I can confirm this for `D16_UNORM` images. The CTS tests fail for `MIN` and `MAX` on my 2016 MBP. I filed a feedback for this, FB8887081. `D24_UNORM` and `D32_FLOAT` work just fine.

> On a related note: does anybody know of something better than this (and the associated pdf):
> https://developer.apple.com/documentation/metal/mtlfeatureset
> 
> to figure out various capabilities of metal.

Sorry, no. It looks like these were supposed to be similar to Direct3D 10.1+'s feature levels, where your card has to support everything implied by the feature level. That's what that doc is supposed to track for Metal. MoltenVK has to basically hardcode this knowledge into itself--see the `MVKPhysicalDeviceMetalFeatures` struct.

**Comment by FilippoLeon** [comment]
Thanks, that is what I was afraid of. This makes using metal's Depth resolve modes different than zero a bit difficult to use, even with direct metal calls. Unless i whitelist each video card individually.
